tests/fuzz/fuzz_data_producer.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ */
+
+/**
+ * Helper APIs for generating random data from input data stream.
```

Comment:
Nit: Can you add a longer description of what this does?
* How does it generate data at a high level?
* What does it do when it is out of bytes?